### PR TITLE
[DI] Fix mypy errors

### PR DESCRIPTION
**Pull request by YalinLi0312**
This PR:
- fixed mypy errors in generated code by regenerating the package
- ignored mypy errors in patch.py
- fixed mypy errors in samples - all are from iterating an optional attribute.
- disabled pylint check in ci as all errors are from generated code, requested generator to fix in issue https://github.com/Azure/autorest.python/issues/2148#issuecomment-1885801941

### CODE REVIEW

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_operations.py [hunk 1]
```diff
@@ -571,6 +572,730 @@ def begin_analyze_document(
...
+        Example:
+            .. code-block:: python
+
+                # JSON input template you can fill out and use as your body input.
+                analyze_request = {
+                    "base64Source": bytes("bytes", encoding="utf-8"),  # Optional. Base64
+                      encoding of the document to analyze.  Either urlSource or base64Source must be
+                      specified.
+                    "urlSource": "str"  # Optional. Document URL to analyze.  Either urlSource or
+                      base64Source must be specified.
```
Comment (kristapratico) [inline sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_operations.py:585]: @iscai-msft document intelligence has models, should it be generating with the request/response templates in the docstrings?

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_version.py [hunk 2]
```diff
@@ -6,4 +6,4 @@
 # Changes may cause incorrect behavior and will be lost if the code is regenerated.
 # --------------------------------------------------------------------------
 
-VERSION = "1.0.0b2"
+VERSION = "1.0.0b1"
```
Comment (kristapratico) [inline sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_version.py:9]: This should be b2 right?

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/models/_models.py [hunk 3]
```diff
@@ -1726,21 +1727,21 @@ class DocumentList(_model_base.Model):
...
-    :ivar items: Items in the list. Required.
-    :vartype items: list[~azure.ai.documentintelligence.models.DocumentListItem]
+    :ivar items_property: Items in the list. Required.
+    :vartype items_property: list[~azure.ai.documentintelligence.models.DocumentListItem]
     """
 
     spans: List["_models.DocumentSpan"] = rest_field()
     """Location of the list in the reading order concatenated content. Required."""
-    items: List["_models.DocumentListItem"] = rest_field()
+    items_property: List["_models.DocumentListItem"] = rest_field(name="items")
```
Comment (kristapratico) [inline sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/models/_models.py:1736]: The generator had to rename this because `items` conflicts with the base model MutableMapping `items`. Are you okay with this name or do you want to rename it in the TypeSpec?

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/aio/sample_analyze_addon_barcodes_async.py [hunk 4]
```diff
@@ -71,17 +71,18 @@ async def analyze_barcodes():
                 features=[DocumentAnalysisFeature.BARCODES],
                 content_type="application/octet-stream",
             )
-        result = await poller.result()
+        result: AnalyzeResult = await poller.result()
```
Comment (kristapratico) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/aio/sample_analyze_addon_barcodes_async.py:74]: result should already be inferred as an `AnalyzeResult`. Did you add this for clarity?

File: sdk/documentintelligence/azure-ai-documentintelligence/setup.py [hunk 5]
```diff
@@ -13,7 +13,7 @@
 
 
 PACKAGE_NAME = "azure-ai-documentintelligence"
-PACKAGE_PPRINT_NAME = "Azure AI Document Intelligence"
+PACKAGE_PPRINT_NAME = "Azure Ai Documentintelligence"
```
Comment (kristapratico) [inline sdk/documentintelligence/azure-ai-documentintelligence/setup.py:16]: It seems like we should revert this change

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_version.py (hunk 2 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_version.py:9]: Yes, it should be b2. Weird, it auto update after regenerating. cc: @iscai-msft

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/aio/sample_analyze_addon_barcodes_async.py (hunk 4 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/aio/sample_analyze_addon_barcodes_async.py:74]: Yes, no errors in mypy for it.

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/models/_models.py (hunk 3 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/models/_models.py:1736]: Let's keep name for now, I'll ask the rename in v-team.

File: sdk/documentintelligence/azure-ai-documentintelligence/setup.py (hunk 5 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/setup.py]: Reverted.

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_version.py (hunk 2 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_version.py]: Reverted.

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/models/_models.py (hunk 3 above)
Comment (kristapratico) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/models/_models.py:1736]: You might want to add this rename to the changelog in the meantime, so we don't forget

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_operations.py (hunk 1 above)
Comment (kristapratico) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_operations.py:585]: Let's follow up with Izzy offline about this

**Review by kristapratico — APPROVED** [review]

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/models/_models.py (hunk 3 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/models/_models.py:1736]: Yup, have mentioned it in changelog.

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_operations.py (hunk 1 above)
Comment (iscai-msft) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_operations.py:585]: We had decided that we were going to cc @johanste and @lmazuel, because we want to generate with sample input. We will probably change what our sample looks like in the future, but we do want to expose sample input to users
